gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/hvdc/HvdcLinkProcessor.java

```diff
@@ -111,93 +130,136 @@ private static void createVscStation(Network network, HvdcCreationParameters cre
+                    copyInformationFromInitialNetwork(optionalLoad1, optionalGen1, optionalLoad2, optionalGen2, optionalLine, hvdcInformation);
+                }
+            });
+        }
     }
 
-    private static void connectEquivalentAcLine(Network network, HvdcCreationParameters creationParameters,
-                                                Map<NetworkElement, Collection<String>> missingElementsMap) {
-        HvdcLine hvdcLine = network.getHvdcLine(creationParameters.getId());
-        getOptionalLine(network, creationParameters.getEquivalentAcLineId(), missingElementsMap)
-                .ifPresent(ol -> {
-                    HvdcAngleDroopActivePowerControl angleDroopActivePowerControl = hvdcLine.getExtension(HvdcAngleDroopActivePowerControl.class);
-                    if (angleDroopActivePowerControl != null && angleDroopActivePowerControl.isEnabled()) {
-                        if (hvdcLine.getConverterStation1().getTerminal().isConnected()) {
-                            ol.getTerminal1().connect();
-                        }
-                        if (hvdcLine.getConverterStation2().getTerminal().isConnected()) {
-                            ol.getTerminal2().connect();
-                        }
-                    }
-                });
+    private static void copyInformationFromInitialNetwork(Optional<Load> optionalLoad1, Optional<Generator> optionalGen1, Optional<Load> optionalLoad2, Optional<Generator> optionalGen2, Optional<Line> optionalAcLine, HvdcInformation hvdcInformation) {
+        if (optionalLoad1.isPresent()) {
+            optionalLoad1.get().setP0(hvdcInformation.getSide1LoadP());
+            if (hvdcInformation.isSide1LoadConnected()) {
+                optionalLoad1.get().connect();
+            }
+        }
+
+        if (optionalLoad2.isPresent()) {
+            optionalLoad2.get().setP0(hvdcInformation.getSide2LoadP());
+            if (hvdcInformation.isSide2LoadConnected()) {
+                optionalLoad2.get().connect();
+            }
+        }
+
+        if (optionalGen1.isPresent()) {
+            optionalGen1.get().setTargetP(hvdcInformation.getSide1GeneratorTargetP());
+            if (hvdcInformation.isSide1GeneratorConnected()) {
+                optionalGen1.get().connect();
+            }
+        }
+
+        if (optionalGen2.isPresent()) {
+            optionalGen2.get().setTargetP(hvdcInformation.getSide2GeneratorTargetP());
+            if (hvdcInformation.isSide2GeneratorConnected()) {
+                optionalGen2.get().connect();
+            }
+        }
+
+        if (optionalAcLine.isPresent()) {
+            if (hvdcInformation.isAcLineTerminal1Connected()) {
+                optionalAcLine.get().getTerminal1().connect();
+            }
+            if (hvdcInformation.isAcLineTerminal2Connected()) {
+                optionalAcLine.get().getTerminal2().connect();
+            }
+        }
```

Comment:
In this method, maybe replace all 'if(optionalLoad.isPresent()) { optionalLoad.get() ....}'
with 'optionalLoad.ifPresent( load -> { ...})' ?